packages/blocks/src/matrix-question/cell.js

```diff
@@ -10,17 +10,28 @@ import { FormCheckbox } from '../components';
 import { MatrixCell } from './styles';
 
 const Cell = ( { column, multipleChoice, questionClientId, row } ) => {
-	const { inputProps } = useField( {
-		name: `q_${ questionClientId }[${ row.clientId }]${
+	const { fieldValue, onUpdate } = useField( {
+		fieldName: `q_${ questionClientId }[${ row.clientId }]${
 			multipleChoice ? '[]' : ''
 		}`,
-		type: multipleChoice ? 'checkbox' : 'radio',
-		value: column.clientId,
+		isMultiSelect: multipleChoice,
 	} );
 
+	const isSelected = multipleChoice
+		? fieldValue.includes( column.clientId )
+		: fieldValue === column.clientId;
+
 	return (
 		<MatrixCell as="label">
-			<FormCheckbox { ...inputProps } />
+			<FormCheckbox
+				checked={ isSelected }
+				isMultiSelect={ multipleChoice }
+				onChange={ ( event ) =>
```

Comment:
The `event -> value` mapping should be done in our input components. So here it's just `onChange={ onUpdate }` or `onChange={ setFieldValue }`.